ui/app/mirrors/create/tablemapping.tsx

```diff
@@ -34,51 +34,40 @@ const TableMapping = ({
   setSchema,
 }: TableMappingProps) => {
   const [allSchemas, setAllSchemas] = useState<string[]>();
-  const [allTables, setAllTables] = useState<string[]>();
   const [tableColumns, setTableColumns] = useState<
     { tableName: string; columns: string[] }[]
   >([]);
   const [loading, setLoading] = useState(false);
 
   const handleAddRow = (source: string) => {
-    setRows([...rows, { source, destination: source, partitionKey: '' }]);
-  };
-
-  const handleRemoveRow = (source: string) => {
     const newRows = [...rows];
     const index = newRows.findIndex((row) => row.source === source);
-    if (index >= 0) newRows.splice(index, 1);
+    if (index >= 0) newRows[index].selected = true;
```

Comment:
Problem is `[...rows]` is a shallow copy. So instead of `selected = true` you need to do `newRows[index] = { ...newRows[index], selected: true }`

Alternatively, use `WeakSet`. Instead of setting newRows, you'd have a `WeakSet<Row>` & at these points you'd create a fresh one & have `setSelectedRows(newWeakSet)` instead of `setRows`. Then code would check `weakset.has(row)` to know what's selected

Reply:
Decided to go with the first option and performing setRows(newRows)